docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,48 +8,140 @@ title: Authorization
+- OAuth 2.0 Demonstrating Proof of Possession (DPoP), for supporting clients
+  ([RFC 9449](https://datatracker.ietf.org/doc/html/rfc9449))  
+- Resource Indicators for OAuth 2.0
+  ([RFC 8707](https://www.rfc-editor.org/rfc/rfc8707.html))
+
+This specification **DOES NOT** require MCP server developers to implement their own authorization server (AS)
+and instead relies on developers adopting dedicated and well-tested AS implementations.
+
+This specification also enables existing deployments to leverage their current infrastructure.
+As a side-effect, this also simplifies consent flows, and services that have multiple MCP Servers
+won't require individual registration per client:server pair.
 
 ## 2. Authorization Flow
 
 ### 2.1 Overview
 
-1. MCP auth implementations **MUST** implement OAuth 2.1 with appropriate security
-   measures for both confidential and public clients.
+1. Authorization servers **MUST** implement OAuth 2.1 with appropriate security measures
+   (see [RFC 9700](https://datatracker.ietf.org/doc/html/rfc9700)) for both confidential and
+   public clients.
+
+2. Authorization servers **SHOULD** support the OAuth 2.1 Dynamic Client Registration Protocol
+   ([RFC 7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+
+3. Authorization servers **SHOULD** implement and MCP clients **MUST** consume OAuth 2.0 Authorization
+   Server Metadata ([RFC 8414](https://datatracker.ietf.org/doc/html/rfc8414)).
+
+4. MCP servers **MUST** implement and support Auth 2.0 Protected Resource Metadata
+   ([RFC DRAFT](https://datatracker.ietf.org/doc/draft-ietf-oauth-resource-metadata/13/)).
+
+### 2.2 Client OAuth 2.1 Authorization
+
+When authorization is required and not yet proven by the client, servers **MUST** respond with
+`HTTP 401 Unauthorized`.
+
+A `HTTP 401 Unauthorized` response **MUST** include a `WWW-Authenticate` header with
+the URL to the protected resource metadata document hosted by the MCP Server in
+the following format:
+
+```http
+WWW-Authenticate: Bearer resource_metadata="https://resource.example.com/.well-known/oauth-protected-resource",
+     scope="mcp:read mcp:write"
+```
+
+A MCP server **MUST** host its Protected Resource Metadata (PRM) document, as outlined in
+[Section 3 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-3).
+
+A MCP server **MUST** return a set of scopes in the `scope` parameter within the `WWW-Authenticate` header
+in the initial handshake as well as in subsequent responses that require incremental authorization.
+
+A MCP server **MUST** include its AS configuration in the PRM document as outlined in
+[Section 2 of the RFC draft](https://www.ietf.org/archive/id/draft-ietf-oauth-resource-metadata-13.html#section-2).
+
+A MCP client **MUST** issue a `GET` request to the URL in `resource_metadata` to get information about
+the AS.
+
+A MCP client **MUST** verify that the `resource` value in the PRM document matches the MCP server
```

Comment:
Resouce values include a protocol as well as the FQDN. Should the protocol also be matched?

Reply:
Good catch - updated.